_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_vaapi.cpp

```diff
@@ -445,6 +445,43 @@ mfxStatus SetFrameRate(
     return MFX_ERR_NONE;
 }
 
+mfxStatus SetMaxSliceSize(
+    MfxVideoParam const & par,
+    VADisplay    vaDisplay,
+    VAContextID  vaContextEncode,
+    VABufferID & maxSliceSizeBuf_id)
+{
+    VAStatus vaSts;
+    VAEncMiscParameterBuffer *misc_param;
+    VAEncMiscParameterMaxSliceSize *maxSliceSize_param;
+
+    MFX_DESTROY_VABUFFER(maxSliceSizeBuf_id, vaDisplay);
```

Comment:
do not use this macro, it doesn't check returned status from ```vaDestroyBuffer```
You can use following function for example:
https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h#L53